src/main/java/ch/uzh/ifi/hase/soprafs23/entity/User.java

```diff
@@ -94,37 +99,48 @@ public void setStatus(UserStatus status) {
         this.status = status;
     }
 
-    public String getAllergies() {
+    public Set<String> getAllergiesSet() {
         return allergies;
     }
 
-    public void setAllergies(String allergies) {
+    public void setAllergiesSet(Set<String> allergies) {
         this.allergies = allergies;
     }
 
-    public String getFavoriteCuisine() {
-        return favoriteCuisine;
+    public void addAllergy(String allergy) {
+        if (allergies == null) {
+            allergies = new HashSet<>();
+        }
+        allergies.add(allergy);
     }
 
+    public void removeAllergies() {
```

Comment:
would be nicer if the naming was more consistent, so getAllergies, setAllergies, addAllergy, removeAllergy, also I'm not sure the setAllergies is necessary, I would have replaced it with the add and remove

Reply:
I like it that we have clearly stated that its a set. But I alligned it a bit to have now: getAllergiesSet, setAllergiesSet, addAllergy, removeAllergy